projects/vdk-plugins/vdk-structlog/src/vdk/plugin/structlog/formatters.py

```diff
@@ -153,8 +155,12 @@ def create_formatter(logging_format: str, metadata_keys: str) -> [Formatter, Fil
         )
         custom_key_filter = ConsoleMetadataFilter(key_set)
     else:
-        formatter = ConsoleFormatter(
-            fmt=StructlogMetadataBuilder(metadata_keys).build_console_format()
-        )
+        if custom_console_format:
+            format_string = custom_console_format
+        else:
+            format_string = StructlogMetadataBuilder(
+                metadata_keys
+            ).build_console_format()
+        formatter = ConsoleFormatter(fmt=format_string)
```

Comment:
We should probably do some validation on the format string, and disregard it if it doesn't pass. Otherwise a misconfigured format string will cause difficult to debug problems with logs down the line.

Reply:
We talked it over with the team and decided it would be an overkill. And also there is not a common way to do it, i will need to generate a regex which is not really safe since it might not work for every possible option.